httpcore/_async/connection_pool.py

```diff
@@ -201,7 +201,16 @@ async def handle_async_request(
     ) -> Tuple[int, Headers, AsyncByteStream, dict]:
         if url[0] not in (b"http", b"https"):
             scheme = url[0].decode("latin-1")
-            raise UnsupportedProtocol(f"Unsupported URL protocol {scheme!r}")
+            host = url[1].decode("latin-1")
+            if scheme == "":
+                raise UnsupportedProtocol(
+                    f"The request to '://{host}/' is missing either an 'http://' or 'https://' protocol."
```

Comment:
Looks like this string [needs to be split across two lines](https://github.com/encode/httpcore/pull/354/checks?check_run_id=2781908584).